### DISCUSSION TITLE
Offline version does not show the list of Blocks

### DISCUSSION THREAD

**Discussion by sdliew** [BIPES/BIPES#118] [General]
Hello,

BIPES online version works great, thank you for your efforts.

However, the offline version does not show the list of blocks, when Blocks is selected. What appeared are three dots ... on the left of the browser. When the Files menu is clicked, File manager appears as in the online version. Other features work fine, and I am able to connect to ESP32 board through Console.

**Comment by jetannenbaum** [comment]
Rebuild the index_offline.html file.  That should solve your problem.  Conversely, you can download a functional offline version from the offline branch.  Be aware though, that it was only tested with the RP2040 based boards.  Some of the micropython that gets generated may be incorrect for the ESP32 based boards.  Also, not all libraries were converted to run offline.  Take a look at the pylibsBlobs folder for the converted libraries.  If you want to add more, you will need to "blobify" the libraries, load the resultant js in the index.html and index_offline.html file, then add the add the libraries in the lib load section in the code.js file
